packages/gatsby-theme-bodiless/src/dist/GatsbyMobxStore.ts

```diff
@@ -103,6 +93,13 @@ class Item {
     }
   }
 
+  private setStatus(newStatus: ItemStatus) {
+    if (newStatus === ItemStatus.Clean && this.status !== ItemStatus.Flushing) {
```

Comment:
Add a doc here describing the purpose of the check, eg:
```suggestion
    if (newStatus === ItemStatus.Clean && this.status !== ItemStatus.Flushing) {
      // If an update happens while flushing, the status will be set to dirty. In this case
      // we don't want to reset it to clean.
```

or - maybe we'd be better served here by a simple state machine, eg:
```
enum ItemStateEvent {
  UpdateFromBrowser,
  BeginPostData,
  EndPostData,
}

private updateState(event: ItemStateEvent) {
  switch (event) {
    case ItemStateEvent.UpdateFromBrowser:
      this.state = ItemState.Dirty;
      break;
    case ItemStateEvent.BeginPostData:
      this.state = ItemState.Flushing;
      break;
    case ItemStateEvent.EndPostData:
      this.state = this.state === ItemState.Dirty ? ItemState.Dirty : ItemState.Clean;
      break;
  }
}
```
To me this is clearer, but up to you if you want to take the time to refactor it.

Reply:
@wodenx, thanks for your suggestion.
addressed in https://github.com/johnsonandjohnson/Bodiless-JS/pull/52/commits/b91098c4c2e7eb2da36a52190875895e13c185a9